tests/test_dataclasses.py

```diff
@@ -522,16 +522,27 @@ class User:
         id: int
         other: Dict[str, str] = dataclasses.field(default_factory=lambda: {'John': 'Joey'})
 
-    user = User(id=123)
-    assert user.id == 123
-    # assert user.other == {'John': 'Joey'}
-    fields = user.__pydantic_fields__
+    @pydantic.dataclasses.dataclass
+    class User2:
+        id: int
+        other: Dict[str, str] = pydantic.dataclasses.Field(default_factory=lambda: {'John': 'Joey'})
 
-    assert fields['id'].is_required() is True
-    assert repr(fields['id'].default) == 'PydanticUndefined'
+    user1 = User(id=123)
+    user2 = User2(id=123)
 
-    assert fields['other'].is_required() is False
-    assert fields['other'].default_factory() == {'John': 'Joey'}
+    def validate_user(user):
+        assert user.id == 123
+        assert user.other == {'John': 'Joey'}
+        fields = user.__pydantic_fields__
+
+        assert fields['id'].is_required() is True
+        assert repr(fields['id'].default) == 'PydanticUndefined'
+
+        assert fields['other'].is_required() is False
+        assert fields['other'].default_factory() == {'John': 'Joey'}
+
+    validate_user(user1)
+    validate_user(user2)
```

Comment:
Similar advice - instead of doing this here, just parametrize the test with the field constructor, then you can just run through the validation checks once 👍